notebooks/6_search_in_continuous_spaces/index.md

```diff
@@ -0,0 +1,313 @@
+  * [Differentiable optimization and Non-differentiable optimization](#differentiable-optimization-and-non-differentiable-optimization)
+    + [Cost functions](#cost-functions)
+- [Convexity](#convexity)
+  * [Convex versus non-Convex Optimization Problems](#convex-versus-non-convex-optimization-problems)
+  * [Local and global optimization](#local-and-global-optimization)
+    + [Local optimization](#local-optimization)
+    + [Global optimization](#global-optimization)
+  * [Locally optimal for a convex optimization problem](#locally-optimal-for-a-convex-optimization-problem)
+  * [Hessian being positive semi-definite iff convexity (Theorem one: One dimensional)](#hessian-being-positive-semi-definite-iff-convexity--theorem-one--one-dimensional-)
+  * [Hessian being positive semi-definite iff convexity (Theorem two: Multi-dimensional)](#hessian-being-positive-semi-definite-iff-convexity--theorem-two--multi-dimensional-)
+  * [Examples of Convex Functions](#examples-of-convex-functions)
+- [Gradient](#gradient)
+  * [Gradient Descent](#gradient-descent)
+  * [Challenges with gradient descent](#challenges-with-gradient-descent)
+    + [Local minima and saddle points](#local-minima-and-saddle-points)
+    + [Vanishing and Exploding Gradients](#vanishing-and-exploding-gradients)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+
+#
+#
+
+
+# Introduction
+In real life optimization problems, we mostly face continuous environments which are made of continuous variables such as time and location. Thus, in this lecture note we will discuss some useful search methods in continuous spaces. The typical problems are linear programming (LP) and diverse kinds of nonlinear programming (such as convex programming, quadratic programming or semidefinite programming). In this lecture note we will discuss different types of optimization techniques in continuous spaces, the differences between different types of optimization techniques, how convexity will help us solve optimization problems easier, difference between local and global optimization, gradient descent method and the challenges with finding the best learning rate.
+
+**Example.1** (Weber Point) Given a collection of cities (assume on 2D plane) how can we find the location that minimizes the sum of distances to all cities? We model this problem by denoting the locations of the cities as y(1), …,y(m)  and write the optimization problem as:
+
+![](Aspose.Words.153ba005-881a-4c4b-97e3-6ea90eff943d.001.jpeg)
+
+**Example.2** (Image deblurring and denoising) Given corrupted imageY∈Rm×n, reconstruct the image by solving the optimization:
+
+![](Aspose.Words.153ba005-881a-4c4b-97e3-6ea90eff943d.002.jpeg)
+
+where K \* denotes convolution with a blurring filter.
+
+![](Aspose.Words.153ba005-881a-4c4b-97e3-6ea90eff943d.003.jpeg)
+
+*Figure  SEQ Figure \\* ARABIC 1. reconstructed image after solving the optimization*

```

Comment:
![image](https://user-images.githubusercontent.com/33592516/142764272-682dcb1a-d113-414c-b8ab-171743df32ac.png)
this caption has problem yet